src/bg/user-script-registry.js

```diff
@@ -97,28 +97,49 @@ async function installFromSource(source) {
 
 
 function loadUserScripts() {
-  db.then(db => {
-    let txn = db.transaction([scriptStoreName], "readonly");
-    let store = txn.objectStore(scriptStoreName);
-    let req = store.getAll();
-    req.onsuccess = event => {
-      userScripts = {};
-      event.target.result.forEach(details => {
-        let userScript = new EditableUserScript(details);
-        userScripts[details.uuid] = userScript;
-        if (userScript.evalContentVersion != EVAL_CONTENT_VERSION) {
-          userScript.calculateEvalContent();
-          saveUserScript(userScript);
-        }
-      });
-    };
-    req.onerror = event => {
-      console.error('loadUserScripts() failure', event);
-    };
+  return db.then(db => {
+    return new Promise((resolve, reject) => {
+      let txn = db.transaction([scriptStoreName], "readonly");
+      let store = txn.objectStore(scriptStoreName);
+      let req = store.getAll();
+      req.onsuccess = async event => {
+        userScripts = {};
+        await Promise.all(event.target.result.map(async details => {
+          let userScript = new EditableUserScript(details);
+          userScripts[details.uuid] = userScript;
+          if (userScript.evalContentVersion != EVAL_CONTENT_VERSION) {
+            userScript.calculateEvalContent();
+            await saveUserScript(userScript);
+          }
+        }));
+        resolve();
+      };
+      req.onerror = event => {
+        console.error('loadUserScripts() failure', event);
+        reject(event.target.error);
+      };
+    });
   });
 };
 
 
+function clearObjectStore() {
```

Comment:
This doesn't clear `value-store` databases. At the moment this isn't needed in testing, nor the actual code, but it would be useful for both the testing and the code[1]. I suggest creating a function to delete individual databases in `value-store.js` and then looping through `userScripts` creating a list of Promises that could be `Promise#all()`.

In my branch for [1] I just updated `UserRegistry#uninstall` to include the `value-store.js` cleanup, then to remove all scripts I looped through `userScripts` calling the uninstall function. Somewhat inefficient but ensured that all cleanup code was executed. Which may be important should additional cleanup code be required down the line.

[1] For example import / exporting the database information. Clearing the database before import was an option that I provided.

Reply:
Yes, this Pull Request is just for fixing `indexedDB.deleteDatabase('greasemonkey');` that is not working now.
I think that to replace with complete cleanup code is more preferred too.
Should I remove this commit and wait for your branch to be merged?